### PR TITLE
feat(27215): Add validation for Edge entities (topics, topic filters and tags)

### PR DESCRIPTION
**Pull request by vanch3d**
See https://hivemq.kanbanize.com/ctrl_board/57/cards/27215/details/

The PR adds custom format validation for JSONSchema-based validation and any RJSF-based UI widget used to edit the nme of the tree Edge entities.

The validation is based on adding a custom `format` to the property of a JSONSchema object that must be validated against the syntax rules: 
- MQTT topics, identified by `mqtt-topic`
- MQTT topic filters, identified by `mqtt-topic-filter`
- Device Tags, identified by `mqtt-tag` 

```json
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "properties": {
    "topic": {
      "title": "An MQTT Topic",
      "type": "string",
      "format": "mqtt-topic"
    },
    "tag": {
      "title": "The name of the tag",
      "type": "string",
      "format": "mqtt-tag"
    }
  }
}

```

The validation rules are ports of the rules from the HiveMQ Community Edition, see  https://github.com/hivemq/hivemq-community-edition/blob/eb7c194bd68bdec822c23f34c5ff6aa6751ae0fa/src/main/java/com/hivemq/util/Topics.java

The integration with RJSF-based forms is complete as long as the `format` is properly defined in the JSONSchema; every existing widget such as text input, auto-complete select, will apply the relevant rules to the item. 

The user feedback is greatly improved by delivering case-by-case error messages, rather than the regular expression or the tautological message  `must match format "mqtt-tag"`

Adding the custom formats also removes the warning triggered on the `dev console`

```
unknown format "mqtt-tag" ignored in schema at path "#/properties/items/items/properties/tag" 
```

### Out-of-scope
- If the custom `format` is not added to a property that requires validation, the string is expected to be of any kind and no validation will apply (see  "Tag name of the Destination" in the screenshots below"). Reviewing the JSONSchema of the various config elements will need to be done in separate tickets.

### Before
screenshot-localhost_3000-2024_11_13-18_17_55

screenshot-localhost_3000-2024_11_13-18_17_04 (1)

screenshot-localhost_3000-2024_11_13-18_19_02

### After
screenshot-localhost_3000-2024_11_13-18_19_41

screenshot-localhost_3000-2024_11_13-18_20_06

screenshot-localhost_3000-2024_11_13-18_20_41

### CODE REVIEW

File: hivemq-edge/src/frontend/src/components/rjsf/Form/validation.utils.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        !isSharedSubscription ||
+        lastChar != SHARED_SUBSCRIPTION_DELIMITER
+      ) {
+        return i18n.t('rjsf.customFormats.validation.noSingleLevelString', { ns: 'components' })
+      }
+    }
+    //Let's check if the + sign is followed by a
+    if (lastChar == '+' && currentChar != '/') {
+      return i18n.t('rjsf.customFormats.validation.noSingleLevelFinal', { ns: 'components' })
+    }
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/components/rjsf/Form/validation.utils.ts:99]:
```suggestion
    if (currentChar == '+' && lastChar != '/') {
      if (
        sharedSubscriptionDelimiterCharCount != 2 ||
        !isSharedSubscription ||
        lastChar != SHARED_SUBSCRIPTION_DELIMITER
      ) {
        return i18n.t('rjsf.customFormats.validation.noSingleLevelString', { ns: 'components' })
      }
      
      return i18n.t('rjsf.customFormats.validation.noSingleLevelFinal', { ns: 'components' })
    }
    
```

**Review by oli-hivemq — APPROVED** [review]
LGTM 🦣

File: hivemq-edge/src/frontend/src/components/rjsf/Form/validation.utils.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      aa.message = validationTopic(aa.data as string)
+      continue
+    }
+    if (aa.schema === 'mqtt-topic-filter') {
+      aa.message = validationTopicFilter(aa.data as string)
+    }
+    if (aa.schema === 'mqtt-tag') {
+      aa.message = validationTag(aa.data as string)
+    }
+  }
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/components/rjsf/Form/validation.utils.ts:124]:
```suggestion
  for (const error of errors) {
    if (error.keyword !== 'format') continue
    if (error.schema === 'mqtt-topic') {
      error.message = validationTopic(error.data as string)
      continue
    }
    if (error.schema === 'mqtt-topic-filter') {
      error.message = validationTopicFilter(error.data as string)
    }
    if (error.schema === 'mqtt-tag') {
      error.message = validationTag(error.data as string)
    }
  }
```

**Review by h2xd — APPROVED** [review]
LGTM! 🐊 

Added couple of suggestions

**Pull request merged**